web3/process.go

```diff
@@ -161,48 +199,131 @@ func (c *Contracts) SetProcessTransition(processID, proof, inputs []byte, blobsS
 
+	data, err := processABI.Pack("submitStateTransition", pid, proof, inputs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to pack data: %w", err)
+	}
+
+	return c.buildDynamicFeeTx(ctx, c.ContractsAddresses.ProcessRegistry, data, nonce)
+}
+
+// buildBlobTransition builds a blob state transition transaction with the given nonce
+func (c *Contracts) buildBlobTransition(ctx context.Context, pid [32]byte, proof, inputs []byte, blobsSidecar *gtypes.BlobTxSidecar, nonce uint64) (*gtypes.Transaction, error) {
 	processABI, err := c.ProcessRegistryABI()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get process registry ABI: %w", err)
 	}
-	tx, err := c.NewEIP4844Transaction(
+
+	return c.NewEIP4844TransactionWithNonce(
 		ctx,
 		c.ContractsAddresses.ProcessRegistry,
 		processABI,
-		"setProcessResults",
+		"submitStateTransition",
 		[]any{pid, proof, inputs},
 		blobsSidecar,
+		nonce,
 	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create EIP-4844 transaction: %w", err)
+}
+
+func (c *Contracts) SetProcessResults(processID, proof, inputs []byte) (*common.Hash, error) {
+	var pid [32]byte
+	copy(pid[:], processID)
+	ctx, cancel := context.WithTimeout(context.Background(), web3QueryTimeout)
+	defer cancel()
+
+	// Use transaction manager if available for automatic nonce management
+	if c.txManager != nil {
+		// Results are always regular transactions (no blobs)
+		return c.txManager.SendTransactionWithFallback(ctx, func(nonce uint64) (*gtypes.Transaction, error) {
+			return c.buildRegularResults(ctx, pid, proof, inputs, nonce)
+		})
 	}
 
-	if err := c.cli.SendTransaction(ctx, tx); err != nil {
-		return nil, fmt.Errorf("failed to set process results (EIP-4844 Tx): %w", err)
+	// Fallback to old method if transaction manager not initialized
+	autOpts, err := c.authTransactOpts()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create transact options: %w", err)
+	}
+	autOpts.Context = ctx
+	tx, err := c.processes.SetProcessResults(autOpts, pid, proof, inputs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set process results: %w", err)
 	}
 	hash := tx.Hash()
 	return &hash, nil
 }
 
+// buildRegularResults builds a regular (non-blob) results transaction with the given nonce
+func (c *Contracts) buildRegularResults(ctx context.Context, pid [32]byte, proof, inputs []byte, nonce uint64) (*gtypes.Transaction, error) {
+	processABI, err := c.ProcessRegistryABI()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get process registry ABI: %w", err)
+	}
+
+	data, err := processABI.Pack("setProcessResults", pid, proof, inputs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to pack data: %w", err)
+	}
+
+	return c.buildDynamicFeeTx(ctx, c.ContractsAddresses.ProcessRegistry, data, nonce)
+}
+
+// buildDynamicFeeTx builds a standard EIP-1559 transaction with the given nonce
+func (c *Contracts) buildDynamicFeeTx(ctx context.Context, to common.Address, data []byte, nonce uint64) (*gtypes.Transaction, error) {
```

Comment:
This is a "generic build transaction method". Why is in process.go ?

Also, should this thing be done within the txManager? I still don't understand the Architecture you decided to implement, please explain.